pkg/reconciler/testing/jobsource.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"context"
+
+	"github.com/n3wscott/sources/pkg/apis/sources/v1alpha1"
+	"github.com/n3wscott/sources/pkg/reconciler/jobsource/resources"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type JobSourceOption func(*v1alpha1.JobSource)
+
+func NewJobSource(name string, options ...JobSourceOption) *v1alpha1.JobSource {
```

Comment:
I would expect the image or template to be set here or in this file for JobSource

Reply:
You can always

```go
myJobSource := NewJobSource("my-jobsource", func(js *v1alpha1.JobSource) {
    js.Spec.Template = ...
})
```

Perhaps it would be beneficial to have
```go
func WithTemplate(template  v1.PodTemplateSpec) JobSourceOption {
    return func (js *v1alpha1.JobSource) {
        js.Spec.Template = template
    }
}
```

Regardless, none of the tests (right now) need to use the template, so I didn't bother.